### DISCUSSION TITLE
Trigger DelegateCommand without buttons

### DISCUSSION THREAD

**Discussion by achobanov** [PrismLibrary/Prism#2625] [WPF]
I want to trigger a DelegateCommand without having the user click a button. For example - I currently have a search box and a button. In order to find something you need to type the value AND press the search button. Instead I would like the search to perform automatically when the user types the first 3 symbols.

**Comment by achobanov** [comment]
Guess that's a no?

**Comment by sigmarsson** [comment]
https://stackoverflow.com/questions/950770/autocomplete-textbox-in-wpf

**Comment by ghost** [comment]
This may be a necropost, but why not put the method the DelegateCommand points to in the set of the text that bound to the textbox?
`private string _textboxText;`
`    public string TextboxText`
`    {`
`        get => _textboxText;`
`        set { `
`              SetProperty(ref _textboxText, value);`
`             if (_textboxText.length() >= 3)`
`                   MethodDelegatePointsTo();`
`    }`

Just make sure the Binding UpdateSourceTrigger is set to OnPropertyChanged.
Mode can be OneWayToSource or TwoWay

**Reply by achobanov** [reply]
Hi, James. Unfortunately that approach does not work. I don't understand why exactly, but the framework seems to only trigger setters once a DelegateCommand has been invoked. I assume it's related to optimizations, but it seems like it's holding the state in the View object and only once a command is executed does Prism update the ViewModel itself. 

Looking at it now - I'm not exactly using Prism as originally intended. I'm not interested in MVVM, because I've decided to layer my application in a different way. The separation between Views and ViewModels is redundant in my architecture. But I'm also liking Prism and at this point I'm not going to be replacing it.

**Reply by brianlagunas** [reply]
To be clear, prism doesn't update any ViewModels directly. That happens via data binding. As @JamesDLynch stated, you need to set the UpdateSourceTrigger to PropertyChanged in order to invoke the setter on each keystroke.

**Reply by achobanov** [reply]
Oh, I've completely missed that last part about the 'UpdateSourceTrigger'. Thanks, Brian, for pointing that out. I'll try it out.
